helm/templates/_helpers.tpl

```diff
@@ -56,10 +56,10 @@ Calculate heap size
 {{- define "superblocks-agent.heapSize" -}}
 {{- if hasSuffix "Mi" . -}}
 {{- $memoryLimit := trimSuffix "Mi" . -}}
-{{ min 512 (div $memoryLimit 2) }}
```

Comment:
Are we just removing a min of 512Mb for agent?

Reply:
This is taking the minimum between half of the memory allocated and 512, which caps the heap size at 512. We're seeing that in addition to the subprocesses, the main process also requires a large heap space from plugins invocations that return large datasets.